adapters/repos/db/vector/flat/index.go

```diff
@@ -353,6 +420,49 @@ func (index *flat) findTopVectors(heap *priorityqueue.Queue, allow helpers.Allow
 	return nil
 }
 
+// populates given heap with smallest distances and corresponding ids calculated by
+// distanceCalc
+func (index *flat) findTopVectorsCached(heap *priorityqueue.Queue, allow helpers.AllowList, limit int,
+	vectorBQ []uint64,
+) error {
+	var id uint64
+	allowMax := uint64(0)
+
+	if allow != nil {
```

Comment:
2 different loops could be implemented: one for allowlist provided case and one for unfiltered search case.
The 1st one instead iterating through entire cache could use allowlist iterator and request cache just for allowed vectors.

1st:
```
for id, found := allow.Next(); found; id, found = allow.Next() {
    vec, err := index.bqCache.Get(context.Background(), id)
    ...
}
```

2nd:
```
for i := 0, len := index.bqCache.Len(); i < len; i++ {
    vec, err := index.bqCache.Get(context.Background(), id)
    ...
}
```